src/output.rs

```diff
@@ -856,4 +923,44 @@ mod tests {
         assert_eq!(results.get("service_name"), None);
         assert_eq!(results.get("service_job_id"), None);
     }
+
+    #[test]
+    fn test_markdown() {
+        let tmp_dir = tempfile::tempdir().expect("Failed to create temporary directory");
+        let file_name = "test_markdown";
+        let file_path = tmp_dir.path().join(&file_name);
+
+        let results = vec![
+            (
+                PathBuf::from("foo/bar/a.cpp"),
+                PathBuf::from("foo/bar/a.cpp"),
+                CovResult {
+                    lines: [(1, 10), (2, 11)].iter().cloned().collect(),
+                    branches: BTreeMap::new(),
+                    functions: FxHashMap::default(),
+                },
+            ),
+            (
+                PathBuf::from("foo/bar/b.cpp"),
+                PathBuf::from("foo/bar/b.cpp"),
+                CovResult {
+                    lines: [(1, 0), (2, 10), (4, 10), (5, 0), (7, 0)]
+                        .iter()
+                        .cloned()
+                        .collect(),
+                    branches: BTreeMap::new(),
+                    functions: FxHashMap::default(),
+                },
+            ),
+        ];
+
+        let results = Box::new(results.into_iter());
+        output_markdown(results, Some(&file_path));
+
+        let results = &read_file(&file_path);
+        let expected_path = PathBuf::from("./test/").join(&file_name);
+        let expected = &read_file(&expected_path);
```

Comment:
Since it's just a few lines, could you put the expected output in a string in this test directly?

Reply:
Updated